proj_eclipse/src/main/java/smartmag/models/BoxModel.java

```diff
@@ -144,6 +161,9 @@ private void cambiaProdotto(Prodotto p) {
 	protected void deleteRecord() {
 		if (isSavedInDb()) {
 			record.delete();
+			record = null;
+			instances.remove(box.getIndirizzo());
+			notifyChangeListeners(null);
```

Comment:
Non rimuoverei l'istanza dalla mappa, semplicemente si elimina record (dal db e dell'istanza) e si notifica; isSavedInDb ritornerà _false_
```suggestion
			record = null;
			notifyChangeListeners(null);
```